resources/js/components/fieldtypes/assets/AssetsFieldtype.vue

```diff
@@ -351,6 +352,10 @@ export default {
             return this.config.show_set_alt && ! this.isReadOnly;
         },
 
+        canBrowse() {
+            return this.can('configure asset containers') || this.can('view '+ this.container +' assets')
```

Comment:
This will introduce a breaking change for sites where users only have permission to edit entries and not browse the asset containers.

To workaround this, let's add a toggle to the Assets Fieldtype so it following the asset container's permissions is opt-in.